rpc/src/v1/impls/parity.rs

```diff
@@ -449,4 +450,26 @@ impl<C, M, U, S> Parity for ParityClient<C, M, U> where
 				.map(|res| res.into_iter().map(|res| res.output.into()).collect())
 				.map_err(errors::call)
 	}
+
+	fn submit_work_detail(&self, nonce: H64, pow_hash: H256, mix_hash: H256) -> Result<H256> {
```

Comment:
There's currently some duplicate code for `parity_submitWorkDetail` and `eth_submitWork`. We can put this function into a helper and call it in both `parity_submitWorkDetail` and `eth_submitWork`. Something like this:

In `rpc/src/v1/helpers/...rs`:

```
pub fn submit_work_detail(miner: _, client: _, ...) -> Result<H256> {
  ...
}
```

In `parity.rs`:

```
fn submit_work_detail(&self, nonce: H64, pow_hash: H256, mix_hash: H256) -> Result<H256> {
  helpers::submit_work_detail(self.miner, self.client, nonce, pow_hash, mix_hash)
}
```

In `eth.rs`:

```
fn submit_work(&self, nonce: RpcH64, pow_hash: RpcH256, mix_hash: RpcH256) -> Result<bool> {
  match helpers::submit_work_detail(self.miner, self.client, nonce, pow_hash, mix_hash) {
    Ok(_) => Ok(true),
    Err(_) => Ok(false),
  }
}
```

Reply:
I am trying to make such a change, but I am having trouble.
The question is what type of miner and client should be.
I noticed that in paraity.rs and eth.rs they are all defined as generics and two definitions are different.
I tried to use generics but I encountered an error like this:
```
pub fn submit_work_detail<C: BlockChainClient, M: MinerService>(client: C, miner: M, nonce: H64, pow_hash: H256, mix_hash: H256) -> Result<H256, Error> {...}
```

```
The trait bound `std::sync::Arc<C>: ethcore::miner::BlockChainClient` is not satisfied
rpc/src/v1/impls/eth.rs:785:9
the trait `ethcore::miner::BlockChainClient` is not implemented for `std::sync::Arc<C>`
```

Reply:
OK, I have got it. This works:
```
pub fn submit_work_detail<C: BlockChainClient, M: MinerService>(client: &Arc<C>, miner: &Arc<M>, nonce: H64, pow_hash: H256, mix_hash: H256) -> Result<H256, Error> {...}
```